### PR TITLE
Add visibility flag shell surface

### LINKED ISSUE #554: Add visuability flag to shell surfaces
Shell surfaces need a visuability addidtionally to position and size. This should be availible via `ws_abstract_shell_surface`. We also need an API command

### PR DESCRIPTION
**Pull request by TheNeikos**
This fixes #554

### CODE REVIEW

File: src/compositor/wayland/abstract_shell_surface.c [hunk 1]
```diff
@@ -113,7 +128,7 @@ ws_object_type_id WS_OBJECT_TYPE_ID_ABSTRACT_SHELL_SURFACE = {
     .cmp_callback       = NULL,
 
     .attribute_table    = NULL,
-    .function_table     = NULL,
+    .function_table     = FUNCTIONS,
```
Comment (matthiasbeyer) [inline src/compositor/wayland/abstract_shell_surface.c:131]: Good catch.

File: src/compositor/wayland/abstract_shell_surface.c [hunk 2]
```diff
@@ -344,3 +361,28 @@ cmd_func_set_width_and_height(
...
+        return -EINVAL;
+    }
+
+    struct ws_abstract_shell_surface* self;
+    self = (struct ws_abstract_shell_surface*)
+            ws_value_object_id_get(&stack[0].object_id);
+
+    // `1` is the command string itself
+
+    if (ws_value_get_type(&stack[2].value) != WS_VALUE_TYPE_BOOL) {
```
Comment (matthiasbeyer) [inline src/compositor/wayland/abstract_shell_surface.c:374]: I'd do this check above, before you actually got the reference on `self`, so you do not have to `unref` here!

File: src/compositor/wayland/abstract_shell_surface.c [hunk 3]
```diff
@@ -344,3 +361,28 @@ cmd_func_set_width_and_height(
...
+        ws_object_unref((struct ws_object*) self);
+        return -EINVAL;
+    }
+
+    bool vis = ws_value_bool_get(&stack[2].bool_);
+
+    self->visible = vis;
```
Comment (matthiasbeyer) [inline src/compositor/wayland/abstract_shell_surface.c]: Fit the above two lines into one line?

**Comment by TheNeikos** [comment]
Fixed your annotations

**Comment by matthiasbeyer** [comment]
ACK.

**Comment by neithernut** [comment]
ACK.

**Pull request merged**